sdl_android/src/main/java/com/smartdevicelink/encoder/VirtualDisplayEncoder.java

```diff
@@ -0,0 +1,591 @@
+                }
+
+                @Override
+                public void onOutputFormatChanged(MediaCodec codec, MediaFormat format) {
+                    // nothing to do here
+                }
+            });
+
+            return surf;
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+        return null;
+    }
+
+    /* handle TouchEvent */
+    public void handleTouchEvent(OnTouchEvent touchEvent)
+    {
+        final MotionEvent motionEvent = convertTouchEvent(touchEvent);
+        if (motionEvent != null && presentation.mainView != null) {
+            Log.d(TAG, "Sending touch event!");
+
+            UI_handler.post(new Runnable() {
+                @Override
+                public void run() {
+                    presentation.mainView.dispatchTouchEvent(motionEvent);
+                }
+            });
+        }
+    }
+
+    private MotionEvent convertTouchEvent(OnTouchEvent touchEvent){
+        List<TouchEvent> eventList = touchEvent.getEvent();
+        if (eventList == null || eventList.size() == 0) return null;
+
+        TouchType touchType = touchEvent.getType();
+        if (touchType == null) return null;
+
+        float x;
+        float y;
+
+        TouchEvent event = eventList.get(eventList.size() - 1);
+        List<TouchCoord> coordList = event.getTouchCoordinates();
+        if (coordList == null || coordList.size() == 0) return null;
+
+        TouchCoord coord = coordList.get(coordList.size() - 1);
+        if (coord == null) return null;
+
+        x = coord.getX();
+        y = coord.getY();
+
+        if (x == 0 && y == 0) return null;
+
+        int eventAction = MotionEvent.ACTION_DOWN;
+        long downTime = 0;
+
+        if (touchType == TouchType.BEGIN) {
+            downTime = SystemClock.uptimeMillis();
+            eventAction = MotionEvent.ACTION_DOWN;
+        }
+
+        long eventTime = SystemClock.uptimeMillis();
+        if (downTime == 0) downTime = eventTime - 100;
+
+        if (touchType == TouchType.MOVE)
+            eventAction = MotionEvent.ACTION_MOVE;
+
+        if (touchType == TouchType.END)
+            eventAction = MotionEvent.ACTION_UP;
+
+        return MotionEvent.obtain(downTime, eventTime, eventAction, x, y, 0);
+    }
+
+    private void onStreamDataAvailable(byte[] data, int size) {
+        Log.d(TAG, "onStreamDataAvailable");
+        if (sdlOutStream != null) {
+            try {
+                if (streamWriterThread.getOutputStream() == null) {
+                    streamWriterThread.setOutputStream(sdlOutStream);
+                    Log.d(TAG, "Setting output stream.");
+                }
+
+                Log.d(TAG, "Transmitting data.");
+                streamWriterThread.setByteBuffer(data, size);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        } else {
+            Log.e(TAG, "sdlOutStream is null");
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN)
```

Comment:
If we targetApi on the class we should be able to remove all these calls on the different methods